components/micro-gateway-core/src/main/ballerina/src/gateway/handlers/basic_auth_handler.bal

```diff
@@ -0,0 +1,76 @@
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/auth;
+import ballerina/http;
+import ballerina/runtime;
+import ballerina/stringutils;
+
+# Representation of the basic auth handler
+#
+# + basicAuthProvider - The reference to the basic auth provider instance
+public type BasicAuthHandler object {
+
+    *http:InboundAuthHandler;
+
+    public BasicAuthProvider basicAuthProvider;
+
+    public function __init(BasicAuthProvider basicAuthProvider) {
+        self.basicAuthProvider = basicAuthProvider;
+    }
+
+    # Checks if the request can be authenticated with the Basic Auth header.
+    #
+    # + req - The `Request` instance.
+    # + return - Returns `true` if can be authenticated. Else, returns `false`.
+    public function canProcess(http:Request req) returns @tainted boolean {
+        string authHeader = runtime:getInvocationContext().attributes[AUTH_HEADER].toString().trim();
+        //if '/apikey' resource is called, the handler is hit without the preAuthnFilter.
+        if (stringutils:equalsIgnoreCase("", authHeader)) {
+            authHeader = DEFAULT_AUTH_HEADER_NAME;
+        }
+        if (req.hasHeader(authHeader)) {
+            string headerValue = req.getHeader(authHeader).toLowerAscii();
+            string authScheme = (auth:AUTH_SCHEME_BASIC).toLowerAscii();
+            if (headerValue.startsWith(authScheme)) {
+                printDebug(KEY_AUTHN_FILTER, "Request will be authenticated via basicAuth handler");
+                return true;
+            }
+        }
+        return false;
+    }
+
+    # Authenticates the incoming request with the use of credentials passed as the Basic Auth header.
+    #
+    # + req - The `Request` instance.
+    # + return - Returns `true` if authenticated successfully. Else, returns `false`
+    # or the `AuthenticationError` in case of an error.
+    public function process(http:Request req) returns @tainted boolean | http:AuthenticationError {
+        string authHeader = runtime:getInvocationContext().attributes[AUTH_HEADER].toString().trim();
+        //if '/apikey' resource is called, the handler is hit without the preAuthnFilter.
+        if (stringutils:equalsIgnoreCase("", authHeader)) {
+            authHeader = DEFAULT_AUTH_HEADER_NAME;
+        }
+        string headerValue = req.getHeader(authHeader);
+        string credential = headerValue.substring(5, headerValue.length()).trim();
+        var authenticationResult = self.basicAuthProvider.authenticate(credential);
+        if (authenticationResult is boolean) {
+            return authenticationResult;
+        } else {
+            return prepareAuthenticationError("Failed to authenticate with basic auth handler.", authenticationResult);
+        }
+    }
+};
```

Comment:
Shall we keep an extra line at EOF

Reply:
fix is already merged from some other PR.